tests/utils/test_dict_ops.py

```diff
@@ -3,11 +3,56 @@
 import pytest
 from pydantic import BaseModel
 
-from kpops.utils.dict_ops import generate_substitution, update_nested_pair
+from kpops.utils.dict_ops import (
+    generate_substitution,
+    update_nested,
+    update_nested_pair,
+)
 from kpops.utils.types import JsonType
 
 
 class TestDictOps:
+    @pytest.mark.parametrize(
+        ("list_dic", "expected"),
+        [
+            (
+                [{}, {}, {}, {}],
+                {},
+            ),
+            # deep update nested dicts
+            (
+                [{"k1": {"foo": 1}}, {"k1": {"bar": ""}}, {"k1": {"baz": "2"}}],
+                {"k1": {"foo": 1, "bar": "", "baz": "2"}},
+            ),
+            # do not overwrite different value types, dict in ``original_dict``
+            (
+                [{"k1": {"bar": ""}}, {"k1": 1}],
+                {"k1": {"bar": ""}},
+            ),
+            # do not overwrite different value types, dict in ``other_dict``
+            (
+                [{"k1": 1}, {"k1": {"bar": ""}}, {"k1": {"baz": ""}}],
+                {"k1": 1},
+            ),
+            # do not overwrite None
+            (
+                [{"k1": None}, {"k1": {"foo": "bar"}}, {"k1": {"baz": "bar"}}],
+                {"k1": None},
+            ),
+            # do not overwrite existing values
+            (
+                [{"k1": 1}, {"k1": 2}, {"k1": 3}],
+                {"k1": 1},
+            ),
+        ],
+    )
+    def test_update_nested(
```

Comment:
instead of repeating the test for update pair, please simply assert that update pair is called with the right arguments when update_nested has more than 2 or more args

Reply:
The reason I wrote a test for both of these methods is that both are public. The public method should be tested separately.
I can convert the method `update_nested_pair` into a private method mock it, and verify if it is called.

Reply:
imo there's no need to make it private. I agree about tests, but since `update_nested` doesn't have much functionality on its own it doesn't make sense to apply the whole set of parametrized tests that are used for testing `update_nested_pair`

Reply:
done